File: laboratorio/ZulGurum.java
package laboratorio;

import robocode.*;

public class ZulGurum extends JuniorRobot {
    private Estratega estratega;

    @Override
    public void run() {
        setColors(orange, blue, white, yellow, black);
        estratega = new EstrategaConcreto(this); // hereda de Estratega
        estratega.arranque();                    // arranca como Sniper
        while (true) { doNothing(1); }
    }

    @Override public void onScannedRobot() { estratega.alEscanear(); }
    @Override public void onHitByBullet()  { estratega.alRecibirDano(); }
    @Override public void onHitWall()      { estratega.alChocar(); }
}

File: src/PonceCiamparella/ZulGurum.kt
package PonceCiamparella

import robocode.JuniorRobot
import robocode.HitRobotEvent

open class ZulGurum : JuniorRobot() {
    private var estratega: Estratega? = null

    override open fun run() {
        estratega = Genghis.getInstance()
        estratega?.bind(this)
        estratega?.arranque()
        while (true) {
            doNothing(1)
        }
    }

    override open fun onScannedRobot() {
        estratega?.alEscanear()
    }

    override open fun onHitByBullet() {
        estratega?.alChocar()
    }

    override open fun onHitWall() {
        estratega?.alChocar()
    }

    override open fun onHitRobot() {
        estratega?.alChoqueRobot()
    }
}